### PR TITLE
Indigo Themes: Update List, Progress, and Dropdown

### PR DESCRIPTION
**Pull request by simeonoff**

### CODE REVIEW

**Comment by sbayreva** [comment]
**Linear bar**
1. The space between the label  and the track seems to be bigger than 4px even though it says it is.  The size of the whole component should be 24px as in Angular, here it is 28px:
Screenshot 2024-06-03 at 10 20 09


**Circular progress**
1. I am not sure whether Subtitle 2 it is used as a style for the label
2. The size of the component should be 48px:
Screenshot 2024-06-03 at 10 31 54
Screenshot 2024-06-03 at 10 33 19

**Comment by imincheva** [comment]
**List**
- Since we don't have list items with buttons as secondary actions in the UI kits, only icons, I assume they should have the same gap. If that's the case, then the gap should be fixed to 8px.
- Header text color variable is set correctly, but seems to be disabled or not applied.
- Header text should use Overline text style.
- The height of the header text should be fixed to 16px.
- Paddings on header text should be 6px 12px
- There should be 2px gap between title and subtitle. Currently, I don't see any padding in the code, although visually it seems that there is such, maybe I just can't find it
- Row-gap on the list item (space between start, content, end) should be 8px, not 16px
- The background on hover should have 4px border radius

**Comment by andiesm813** [comment]
> **Linear bar**
> 
> 1. The space between the label  and the track seems to be bigger than 4px even though it says it is.  The size of the whole component should be 24px as in Angular, here it is 28px:
> 
> Screenshot 2024-06-03 at 10 20 09
> **Circular progress**
> 
> 1. I am not sure whether Subtitle 2 it is used as a style for the label
> 2. The size of the component should be 48px:
> 
> Screenshot 2024-06-03 at 10 31 54 Screenshot 2024-06-03 at 10 33 19

Agree to all comments.
Apparently it is Subtitle 2 @sbayreva .... if it wasnt, it would look WAAAY lighter! see below! So this is not an issue. It's good! ✅

image

**Comment by andiesm813** [comment]
### List
Agree to all comments made by @imincheva ... and i'm adding the same comments i added for Angular, which apply here as well.

**LIGHT THEME**

--header-text-color has the correct value in the CSS, however, in the samples, the header seems way darker than grays600. It looks almost as dark as grays900.

**SIZES**
The implemented sizes are incorrect. All the sizes that are implemented for COMFORTABLE correspond to COSY. And the ones implemented for COSY correspond to COMPACT. So the sizes for LARGE need to be defined by me. See below.
These sizes specified below are consistent with the sizes of the TREE component.

Comfy - Large

- The height of the header container should be 32px.
- The height of the single line item should be 32px.
- List Item inner paddings: 8px 16px.

Cosy - Medium

- The height of the header container should be 28px.
- The height of the single line item should be 28px.
- List Item inner paddings: 6px 12px.

Compact - Small

- The height of the header container should be 24px.
- The height of the single line item should be 24px.
- List Item inner paddings: 4px 8px.

**ADDITIONAL COMMENT**
There is no ripple effect on click here, on click. This is the expected behavior. Angular should match this too.

**Comment by AnjiManova** [comment]
**Dropdown**

Currently the default size is Large, I think it should be Medium, but @andiesm813 should confirm.

**Comment by andiesm813** [comment]
@AnjiManova yes, the default should be medium. I see that it is medium now... maybe it was updated since your comment....

**Comment by SisIvanova** [comment]
> > > **Linear bar**
> > > 
> > > 1. The space between the label  and the track seems to be bigger than 4px even though it says it is.  The size of the whole component should be 24px as in Angular, here it is 28px:
> > > 
> > > Screenshot 2024-06-03 at 10 20 09
> > > **Circular progress**
> > > 
> > > 1. I am not sure whether Subtitle 2 it is used as a style for the label
> > > 2. The size of the component should be 48px:
> > > 
> > > Screenshot 2024-06-03 at 10 31 54 Screenshot 2024-06-03 at 10 33 19
> > 
> > 
> > Agree to all comments. Apparently it is Subtitle 2 @sbayreva .... if it wasnt, it would look WAAAY lighter! see below! So this is not an issue. It's good! ✅
> > image
> 
> * Point 1 is verified
> * Point 2 is verified
> * About the font style Subtitle 2, I am not sure whether the font is used as a style, that's why I added the comment. @SisIvanova can you please, confirm whether font style Subtitle 2 is used for the labels of the Linear and Circular Progress Indicators?

@sbayreva As Andie said it is Subtitle 2. You can also see it in the samples applied on the part label/value.
Screenshot 2024-06-10 at 9 44 39 AM

**Comment by SisIvanova** [comment]
> Member

The size should be fixed.

**Comment by andiesm813** [comment]
> > > > **Linear bar**
> > > > 
> > > > 1. The space between the label  and the track seems to be bigger than 4px even though it says it is.  The size of the whole component should be 24px as in Angular, here it is 28px:
> > > > 
> > > > Screenshot 2024-06-03 at 10 20 09
> > > > **Circular progress**
> > > > 
> > > > 1. I am not sure whether Subtitle 2 it is used as a style for the label
> > > > 2. The size of the component should be 48px:
> > > > 
> > > > Screenshot 2024-06-03 at 10 31 54 Screenshot 2024-06-03 at 10 33 19
> > > 
> > > 
> > > Agree to all comments. Apparently it is Subtitle 2 @sbayreva .... if it wasnt, it would look WAAAY lighter! see below! So this is not an issue. It's good! ✅
> > > image
> > 
> > 
> > 
> > * Point 1 is verified
> > * Point 2 is verified
> > * About the font style Subtitle 2, I am not sure whether the font is used as a style, that's why I added the comment. @SisIvanova can you please, confirm whether font style Subtitle 2 is used for the labels of the Linear and Circular Progress Indicators?
> 
> @sbayreva As Andie said it is Subtitle 2. You can also see it in the samples applied on the part label/value. Screenshot 2024-06-10 at 9 44 39 AM

So PROGRESS looks all good then ✅

**Comment by andiesm813** [comment]
> @AnjiManova The dropdown default size should be fixed.

Looks fixed to me! thanks for the update! ✅

**Comment by andiesm813** [comment]
### LIST
Some ofl the fixes requested for this component have not been applied yet, if i'm not mistaken. Maybe it's still in progress?

**Pending fixes**

1. The height of the header text is still not fixed to 16px.
2. The paddings of the list headers are only fixed for large size. The others remain incorrect. In my [previous comments](https://github.com/IgniteUI/igniteui-webcomponents/pull/1224#issuecomment-2146058112) were the correct paddings (list item inner paddings) and expected header sizes, if all is correct (header text fixed height affects the total height of the header).
3. Row-gap on the list item (space between start, content, end) should be 8px, not 16px
4. List Item small inner padding is still not correct. It should be 4px 8px, and it's currently 2px 4px.
5. There is a gap between title and subtitle, but i'm not sure if it's 2px or more.
6. The title should be set to body 2 as well as the subtitle. The only difference between title and subtitle is the foreground color.
7. Gap between secondary actions should be 0.5rem (currently it's 0.375rem)
8. Also not sure which List size is the default size. Currently it's a mixture of Large and Medium. Medium should be the default size. 

image

**Comment by andiesm813** [comment]
> ### LIST
> Some ofl the fixes requested for this component have not been applied yet, if i'm not mistaken. Maybe it's still in progress?
> 
> **Pending fixes**
> 
> 1. The height of the header text is still not fixed to 16px.
> 2. The paddings of the list headers are only fixed for large size. The others remain incorrect. In my [previous comments](https://github.com/IgniteUI/igniteui-webcomponents/pull/1224#issuecomment-2146058112) were the correct paddings (list item inner paddings) and expected header sizes, if all is correct (header text fixed height affects the total height of the header).
> 3. Row-gap on the list item (space between start, content, end) should be 8px, not 16px
> 4. List Item small inner padding is still not correct. It should be 4px 8px, and it's currently 2px 4px.
> 5. There is a gap between title and subtitle, but i'm not sure if it's 2px or more.
> 6. The title should be set to body 2 as well as the subtitle. The only difference between title and subtitle is the foreground color.
> 7. Gap between secondary actions should be 0.5rem (currently it's 0.375rem)
> 8. Also not sure which List size is the default size. Currently it's a mixture of Large and Medium. Medium should be the default size.
> 
> image

Points 1-8 have been verified. So i can say LIST is all good ✅

File: src/components/list/themes/shared/item/list-item.bootstrap.scss [hunk 1]
```diff
@@ -11,11 +11,18 @@ $theme: $bootstrap;
...
-
 [part='title'] {
     @include type-style('body-1');
 }
 
 [part='subtitle'] {
     @include type-style('body-2');
 }
+
+[name='start']::slotted(*) {
```
Comment (SisIvanova) [inline src/components/list/themes/shared/item/list-item.bootstrap.scss:22]: @desig9stein You don't need these margins here. You already have them in the list-item.common

File: src/components/list/themes/shared/item/list-item.fluent.scss [hunk 2]
```diff
@@ -14,3 +14,11 @@ $theme: $fluent;
 [part='title'] {
     @include type-style('caption');
 }
+
+[name='start']::slotted(*) {
```
Comment (SisIvanova) [inline src/components/list/themes/shared/item/list-item.fluent.scss:18]: @desig9stein Same as in bootstrap.

**Review by SisIvanova — DISMISSED** [review]

**Pull request merged**